src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphWorkspaceDAO.scala

```diff
@@ -56,4 +56,18 @@ class GraphWorkspaceDAO extends WorkspaceDAO with GraphDAO {
       case None => false
     }
   }
+
+  val LOCK_PROPERTY = "isLocked"
+
+  override def isLocked(workspaceContext: WorkspaceContext): Boolean = {
+    workspaceContext.workspaceVertex.getProperty(LOCK_PROPERTY)
+  }
+
+  override def lock(workspaceContext: WorkspaceContext): Unit = {
+    workspaceContext.workspaceVertex.setProperty(LOCK_PROPERTY,true)
+  }
+
+  override def unlock(workspaceContext: WorkspaceContext): Unit = {
+    workspaceContext.workspaceVertex.setProperty(LOCK_PROPERTY,false)
+  }
```

Comment:
why do we need special functions for this? shouldn't we just change the property on the Workspace and call save?